kong/tools/string.lua

```diff
@@ -24,16 +31,49 @@ _M.split = pl_stringx.split
 
 --- strips whitespace from a string.
 -- @function strip
-_M.strip = function(str)
-  if str == nil then
+_M.strip = function(value)
+  if value == nil then
     return ""
   end
-  str = tostring(str)
-  if #str > 200 then
-    return str:gsub("^%s+", ""):reverse():gsub("^%s+", ""):reverse()
-  else
-    return str:match("^%s*(.-)%s*$")
+
+  if type(value) ~= "string" then
+    value = tostring(value)
+  end
+
+  if value == "" then
+    return ""
   end
+
+  local len = #value
+  local s = 1
```

Comment:
Could we add a small comment for `s` and `e`, I know they means `start` and `end`, but they are not obvious to a first reviewer.

Reply:
Added.